outputs/openapi/schema.go

```diff
@@ -5,63 +5,80 @@ import (
 	"github.com/ls6-events/astra/astTraversal"
 )
 
-func mapParamToSchema(bindingType astTraversal.BindingTagType, param astra.Param) Schema {
+func mapParamToSchema(bindingType astTraversal.BindingTagType, param astra.Param) (Schema, bool) {
 	if param.IsBound {
 		return mapFieldToSchema(bindingType, param.Field)
 	} else if param.IsArray {
 		itemSchema := mapAcceptedType(param.Field.Type)
 		if !astra.IsAcceptedType(param.Field.Type) {
-			itemSchema = Schema{
-				Ref: makeComponentRef(bindingType, param.Field.Type, param.Field.Package),
+			componentRef, ok := makeComponentRef(bindingType, param.Field.Type, param.Field.Package)
+			if ok {
+				itemSchema = Schema{
+					Ref: componentRef,
+				}
 			}
 		}
 		return Schema{
 			Type:  "array",
 			Items: &itemSchema,
-		}
+		}, true
 	} else if param.IsMap {
 		var additionalProperties Schema
 		if !astra.IsAcceptedType(param.Field.Type) {
-			additionalProperties.Ref = makeComponentRef(bindingType, param.Field.Type, param.Field.Package)
+			componentRef, ok := makeComponentRef(bindingType, param.Field.Type, param.Field.Package)
+			if ok {
+				additionalProperties.Ref = componentRef
+			}
 		} else {
 			additionalProperties = mapAcceptedType(param.Field.Type)
 		}
 		return Schema{
 			Type:                 "object",
 			AdditionalProperties: &additionalProperties,
-		}
+		}, true
 	} else {
-		return mapAcceptedType(param.Field.Type)
+		return mapAcceptedType(param.Field.Type), true
 	}
 }
 
-func mapFieldToSchema(bindingType astTraversal.BindingTagType, field astra.Field) Schema {
+func mapFieldToSchema(bindingType astTraversal.BindingTagType, field astra.Field) (Schema, bool) {
 	if !astra.IsAcceptedType(field.Type) {
-		return Schema{
-			Ref: makeComponentRef(bindingType, field.Type, field.Package),
+		componentRef, ok := makeComponentRef(bindingType, field.Type, field.Package)
+		if ok {
+			return Schema{
+				Ref: componentRef,
+			}, true
 		}
+
+		return Schema{}, false
 	} else {
 		schema := mapAcceptedType(field.Type)
 		if field.Type == "slice" {
 			itemSchema := Schema{
-				Type: mapAcceptedType(field.SliceType).Type,
+				Type: mapAcceptedType(field.Type).Type,
```

Comment:
Isn't field.type here slice? So shouldn't this not work

Reply:
Unless it is array and then the items handle it lower down the line?

Reply:
I think I must have changed it during testing, I have reverted it and re run some of the examples and they don't change. I think the variable get overwritten anyway. We really need to improve our testing.